# GeshaltChallenges
This repo is for those who want to take on challenges to be part of Gestalt ! Enjoy the journey !



***How to get on the journey?***

Due to the fact that we need intensive collaboration on coding sharing, we will model such circumstance addressing challenges within four steps.

Hint: if you are not familiar with GitHub, [GitHub Guide](https://guides.github.com/) might help.
### Step 1. Clone this repo.

### Step 2. Complete as many chanllenges as you can, don't hesitate to ask questions.

### Step 3. Share your results with us, we prefer a link to your github or a cloud drive.


## Challenge 1 - Collecting metadata from publicly  available datasets

Please collect metadata from three metagenome projects: `PRJNA339914`, `PRJEB39223`, `PRJNA561510` and organize them in a manner of [curatedMetagenomicData](https://github.com/waldronlab/curatedMetagenomicDataCuration/blob/master/inst/curated/AsnicarF_2017/AsnicarF_2017_metadata.tsv).

## Challenge 2 - Data visualization from a giant matrix

Please make sense of the massive clinical data from [Human Microbiome Project 2](https://github.com/waldronlab/curatedMetagenomicDataCuration/blob/master/inst/curated/HMP_2019_ibdmdb/HMP_2019_ibdmdb_metadata.tsv) utilizing your visualization expertise.

Hint: the metadata table contains different types of data (e.g. continuous, categorical, binary types etc. ), and the answer is not unique. But we expect that you can maximize skills you already have to mine out maximum information from such a mess.

## Challenge 3 - Digging into genomic data

In this challenge, you will take a genome alignment in FASTA format: [GenomeAln.fna](https://github.com/Geshalt/GeshaltChallenges/blob/main/GenomeAln.fna), and assess five aspects of this genome alignment:

* Gap ratio (the proportion of gaps in the alignment file)
* Length of the genome alignment
* The length of each sequence excluding gaps
* The distribution of SNP (Single-nucleotide polymorphism) rate between pairs of genomes in the alignment. Hint: the SNP rate is defined by the number of variant sites between two genomes divided by the total number of sites when considering only non-gappy sites (both genomes are not gaps at that position).



### Note:
*We do not expect that you have perfect answers for these challenges and there are always countless solutions for the same challenge. Thus, just kill them in the way that you are good at. Last but not least, we hope you can understand it is not an examination that you need to finish with limited time and sources. Questions, debates, arguments, discussions are always welcome during the challenge journey. This will absolutely help us to evaluate your capability better.*
